### ISSUE TITLE
Batch normalization enmeshes two orthogonal functions 

### ISSUE THREAD

**Issue by ad6fp** [mlcommons/training_policies#42]
In the closed division networks that include batch normalization intertwine the use of batch size to amortize latency in hardware with the use of batch size for faster convergence. Future systems are likely to require larger batch sizes to amortize increasing memory latencies. This may not align with the use of batch size for normalization to accelerate convergence.  The closed division should permit a broader range of normalization functions than just batch normalization.

**Comment by bitfort** [comment]
SWG:

Preliminarily, for the closed division we decide against this because it changes the mathematical definition of the model. It is our intention to keep the model definitions consistent across all submissions. If you believe we are misunderstanding the issue or have data to demonstrate this issue, please do reach out. We would welcome you to present results to further inform this decision.

**Comment by ad6fp** [comment]
Victor -

An accelerator that doesn’t require data parallelism to be used to achieve performance (e.g. runs at batch size =1) would suffer
either:

1) a decrease in convergence rate due to BN being effectively disabled at batch = 1
or
2) an increase in activation memory usage by being forced to run at batch > 1

It is unfortunate that normalization is coupled to a parameter that governs HW accelerator performance and memory usage.
To encourage innovation shouldn’t there be a division that uses the same network but allows flexibility in the trining machinery (normalization, loss function, optimizer )
since these are all tightly coupled to accelerator implementation characteristics ?  Refer to my previous issue #33 on the coupling of loss function to numeric representation.
Techniques like deep gradient compression are precluded due to this coupling. The open division is not a sufficient answer since there is no longer any guarantee of comparing
the same network.


Gary Lauterbach
gary@cerebras.net<mailto:gary@cerebras.net>
CTO and Co-Founder
Cerebras Systems




On Aug 2, 2018, at 11:56 AM, Victor Bittorf <notifications@github.com<mailto:notifications@github.com>> wrote:


SWG:

Preliminarily, for the closed division we decide against this because it changes the mathematical definition of the model. It is our intention to keep the model definitions consistent across all submissions. If you believe we are misunderstanding the issue or have data to demonstrate this issue, please do reach out. We would welcome you to present results to further inform this decision.

**Comment by nvpaulius** [comment]
Historically normalization layers were part of the model architecture. Given that different normalizations are not readily swappable (for example, you generally can't just replace batch normalizations with layer normalizations and maintain accuracy), changes to normalization type today become mode changes to fall into the open category. I would say the same about the loss functions and optimizers.

**Comment by ad6fp** [comment]
Except for scaling of the loss function, that will be allowed. Why isn’t that a mode change?
There is already an accuracy threshold, is that insufficient to ensure an equitable comparison?
The closed division mandates the exact same graph be used with exceptions for somewhat ill defined non-determinism and changes
to non-stateful parts of the graph.

I would suggest that training comparisons using the exact same "forward inference only” e.g. Resnet50  graph are relevant and the details of the mechanics
of computing the forward inference only weights should be open to innovation.

Gary Lauterbach
gary@cerebras.net<mailto:gary@cerebras.net>
CTO and Co-Founder
Cerebras Systems




On Aug 3, 2018, at 3:56 PM, nvpaulius <notifications@github.com<mailto:notifications@github.com>> wrote:


Historically normalization layers were part of the model architecture. Given that different normalizations are not readily swappable (for example, you generally can't just replace batch normalizations with layer normalizations and maintain accuracy), changes to normalization type today become mode changes to fall into the open category. I would say the same about the loss functions and optimizers.

**Comment by nvpaulius** [comment]
Loss-scaling doesn't change the algebra of the loss function. The change to the loss function that I don't think would fall within the closed division would be, for example, replacing cross-entropy with L1 distance. Or for optimizers, switching from SGD with momentum to Adam.

**Comment by petermattson** [comment]
I'm believe the WG is recommendation does not require a rules change so I'm closing this issue. Gary, a one-page def on Flex division could still be very interesting.
